src/HomieNode.cpp

```diff
@@ -37,28 +67,34 @@ HomieNode::~HomieNode() {
     return;  // never reached, here for clarity
 }
 
-PropertyInterface& HomieNode::advertise(const char* property) {
-  Property* propertyObject = new Property(property);
+PropertyInterface& HomieNode::advertise(const char* id) {
+  Property* propertyObject = new Property(id);
 
   _properties.push_back(propertyObject);
 
   return _propertyInterface.setProperty(propertyObject);
 }
 
-PropertyInterface& HomieNode::advertiseRange(const char* property, uint16_t lower, uint16_t upper) {
-  Property* propertyObject = new Property(property, true, lower, upper);
-
-  _properties.push_back(propertyObject);
-
-  return _propertyInterface.setProperty(propertyObject);
+SendingPromise& HomieNode::setProperty(const String& property) const {
+  Property* iProperty = this->getProperty(property);
+  if (iProperty) {
+    if (iProperty->isRetained())
+      return Interface::get().getSendingPromise().setNode(*this).setProperty(property).setQos(1).setRetained(true);
+    else
+      return Interface::get().getSendingPromise().setNode(*this).setProperty(property).setQos(1);
+  }
 }
```

Comment:
Here a reference to SendingPromise MUST be returned. For now, this result in a crash if you want to set a Property that has not be configured.

Solutions:

1. Change signature of HomieNode::setProperty to `SendingPromise *HomieNode::setProperty(...)` and return NULL --> caller can check for NULL
2. Create error message and `abort()` --> Homie-ESP8266 enforces that you can only send to advertised properties
3. Return a generic SendingPromise& to be able to send on arbitrary topics (old behaviour. Maybe that is what you 

wanted to do and just messed with the nested `if`s?

I suggest to stick to the Homie-2.0 behaviour for now (third solution)

Reply:
Just change the line 81 und 82 to one line:
```C++
if (iProperty && iProperty->isRetained())
```